### ISSUE TITLE
Accessibility issue in notification widget

### ISSUE THREAD

**Issue by sarahelsaig** [OrchardCMS/OrchardCore#16594]
### Describe the bug

I was running accessibility check on the site with several modules enabled. Everything is fine, except the notifications widget.
![image](https://github.com/user-attachments/assets/cafcf3ba-1520-47bd-8310-23bee4338172)

This could be related to the changes in #15231, though that's just a guess based on the file history of [UserNotificationNavbar.cshtml](https://github.com/OrchardCMS/OrchardCore/blob/3464dba26164099d215efab8e53c7a951623a058/src/OrchardCore.Themes/TheTheme/Views/UserNotificationNavbar.cshtml).
The accessibility report is attached below.

### Orchard Core version

Latest `main`.

### To Reproduce
1. Set up the site.
2. Enable the Notifications feature.
3. Use [`axe`](https://www.npmjs.com/package/@axe-core/cli) to evaluate the home page.

### Expected behavior

The dropdown element should have an appropriate `title`, `aria-label` or `aria-labelledby` attribute.

### Logs

> 1: Links must have discernible text
> Description: Ensures links have discernible text
> Help: Links must have discernible text
> Help URL: https://dequeuniversity.com/rules/axe/4.9/link-name?application=axe-core-selenium-nuget
> Impact: serious
> Tags: cat.name-role-value, wcag2a, wcag244, wcag412, section508, section508.22.a, TTv5, TT6.a, EN-301-549, EN-9.2.4.4, EN-9.4.1.2, ACT
> Element(s):
> Html:
> 
> <a type="button" class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false">
> Selector:
> 
> a[data-bs-auto-close="outside"]
> To solve:
> 
> Fix all of the following issues:
> 
>     SERIOUS: Element is in tab order and does not have accessible text
> 
> Fix at least one of the following issues:
> 
>     SERIOUS: Element does not have text that is visible to screen readers
>     SERIOUS: aria-label attribute does not exist or is empty
>     SERIOUS: aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
>     SERIOUS: Element has no title attribute

**Comment by DrewBrasher** [comment]
If no one is working on this issue I will work on it and create a PR

**Comment by MikeAlhayek** [comment]
@DrewBrasher please submit a PR

**Comment by DrewBrasher** [comment]
I tried adding an aria-label to the notification button but my change isn't showing up in the html when I run it. I'm thinking this is because the menus are cached. Do I need to disable menu caching and if so how do I do that?

**Comment by DrewBrasher** [comment]
Disregard that last question. Thanks to @MikeAlhayek's (https://github.com/OrchardCMS/OrchardCore/pull/16875#issuecomment-2408742743), I see now that the TheTheme is overriding this view.
